core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,165 @@ const createButtonForAction = (action, index) => {
+              </DialogHeader>
+            )}
+
+            <DialogBody
+              ref={this.childrenRef}
+              id="dialog-description"
+              {...Automation('dialog.body')}
+            >
+              {props.children}
+            </DialogBody>
+
+            {props.actions && (
+              <DialogFooter {...Automation('dialog.footer')}>
+                <ButtonGroup>{props.actions.map(createButtonForAction)}</ButtonGroup>
+              </DialogFooter>
+            )}
+          </DialogBox>
+        </FocusTrap>
+      </Overlay>
+    )
+  }
+}
+
+const DialogBox = styled.div`
   position: relative;
-  overflow: hidden;
-  width: ${props => props.width}px;
-  background: ${colors.base.white};
+  max-height: calc(100vh - (${spacing.xlarge} * 2));
+  display: flex;
+  flex-direction: column;
+  background-color: ${colors.base.white};
   border-radius: 3px;
   box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
 `
 
-const DialogTitleBar = styled.div`
-  display: flex;
+const DialogClose = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  
+  /* Overwirtes the color of the icons */
+  ${Button.Element} {
+    &, i, svg, path {
+      color: ${colors.base.black};
+       fill: ${colors.base.black};
+    }
+  }
+`
+
+const DialogHeader = styled.header`
   position: relative;
-  overflow: hidden;
-  padding: ${spacing.small};
-  padding-bottom: 0;
+  padding: ${spacing.small} ${spacing.large} ${spacing.xsmall} ${spacing.large};
   color: ${colors.text.default};
-  font-weight: ${fonts.weight.medium};
+  word-break: break-word;
+  white-space: pre-wrap;
+  word-wrap: break-word;
   text-align: center;
-  span {
-    flex: 1;
-  }
-  & ${Icon.Element} {
-    flex: none;
-    cursor: pointer;
-  }
+`
+
+const DialogTitle = styled(Heading)`
+  font-weight: ${fonts.weight.medium};
+  font-size: 1em;
+  margin: 0;
 `
 
 const DialogBody = styled.div`
-  padding: ${spacing.medium} ${spacing.large};
+  padding: ${spacing.small} ${spacing.medium} ${spacing.large} ${spacing.medium};
+  flex: 1 1 auto;
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+  overscroll-behavior: contain;
+  word-break: break-word;
+  white-space: -moz-pre-wrap;
+  word-wrap: break-word;
+
+  /* Clears the margin of the last item of the body */
+  > * {
+    margin-bottom: 0;
+  }
 `
 
-const DialogFooter = styled.div`
+const DialogFooter = styled.footer`
   display: flex;
+  flex: 0 0 auto;
   justify-content: center;
   padding: ${spacing.small};
   border-top: 1px solid ${colors.base.grayLight};
 `
 
 Dialog.Action = DialogAction
-Dialog.Element = DialogElement
-
+Dialog.Element = DialogBox
 Dialog.propTypes = {
   actions: PropTypes.arrayOf(
     PropTypes.oneOfType([PropTypes.instanceOf(DialogAction), PropTypes.element])
-  ).isRequired,
-  title: PropTypes.string.isRequired,
-  width: PropTypes.number,
-  onClose: PropTypes.func
+  ),
+  title: PropTypes.string,
+  titleElement: PropTypes.oneOf([1, 2, 3, 4]),
```

Comment:
We can document the prop types with comments

Example: https://github.com/auth0/cosmos/blob/master/core/components/atoms/button/button.js#L292-L322